testing/src/main/java/io/stargate/it/http/RestApiv2Test.java

```diff
@@ -2286,6 +2286,158 @@ public void deleteColumnPartitionKey() throws IOException {
     assertThat(response.getDescription()).isNotEmpty();
   }
 
+  @Test
+  public void createUdt() throws IOException {
+    createKeyspace(keyspaceName);
+
+    String udtString =
+        "{\"name\": \"udt1\", \"fieldDefinitions\":[{\"name\":\"firstname\",\"typeDefinition\":\"text\"}, {\"name\":\"birthdate\",\"typeDefinition\":\"date\"}]}";
+
+    RestUtils.post(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+        udtString,
+        HttpStatus.SC_CREATED);
+
+    // throws error because same name, but ifNotExists = false
+    String response =
+        RestUtils.post(
+            authToken,
+            String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+            udtString,
+            HttpStatus.SC_BAD_REQUEST);
+
+    String typeName = "udt1";
+
+    Error error = objectMapper.readValue(response, Error.class);
+    assertThat(error.getCode()).isEqualTo(HttpStatus.SC_BAD_REQUEST);
+    assertThat(error.getDescription())
+        .isEqualTo(
+            String.format(
+                "Bad request: A type named \"%s\".%s already exists", keyspaceName, typeName));
+
+    // don't create and don't throw exception because ifNotExists = true
+    udtString =
+        "{\"name\": \"udt1\", \"ifNotExists\": true, \"fieldDefinitions\":[{\"name\":\"firstname\",\"typeDefinition\":\"text\"}]}";
+    RestUtils.post(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+        udtString,
+        HttpStatus.SC_CREATED);
+  }
+
+  @Test
+  public void dropUdt() throws IOException {
```

Comment:
Might be good to add a test for dropping a type that's in use.

Reply:
Done.